bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/util/ModelUtil.java

```diff
@@ -99,6 +118,58 @@ public static void verifyNoDuplicateSignalStartEvents(
     verifyNoDuplicatedEventDefinition(definitions, errorCollector);
   }
 
+  public static void verifyNoDuplicatedLinkCatchEvents(
+      final ModelElementInstance element, final Consumer<String> errorCollector) {
+
+    final List<EventDefinition> definitions = getEventDefinitionsForLinkCatchEvents(element);
+
+    verifyNoDuplicatedEventDefinition(definitions, errorCollector);
+  }
+
+  public static void verifyPairsLinkEvents(
+      final ModelElementInstance element, final Consumer<String> errorCollector) {
+
+    // get link catch events definition
+    final List<EventDefinition> linkCatchEvents = getEventDefinitionsForLinkCatchEvents(element);
+
+    // get link throw events definition
+    final List<EventDefinition> linkThrowEvents = getEventDefinitionsForLinkThrowEvents(element);
+
+    // get link catch events names
+    final Stream<String> catchEventNames =
+        getEventDefinition(linkCatchEvents, LinkEventDefinition.class)
+            .filter(def -> def.getName() != null && !def.getName().isEmpty())
+            .map(LinkEventDefinition::getName);
+
+    // get link throw events names
+    final Stream<String> throwsEventNames =
+        getEventDefinition(linkThrowEvents, LinkEventDefinition.class)
+            .filter(def -> def.getName() != null && !def.getName().isEmpty())
+            .map(LinkEventDefinition::getName);
+
+    // get distinct link catch events names
+    final List<String> linkCatchList = catchEventNames.distinct().collect(Collectors.toList());
+
+    // get distinct link throw events names
+    final List<String> linkThrowList = throwsEventNames.distinct().collect(Collectors.toList());
+
+    // intersection with linkCatchList and linkThrowList
+    final List<String> intersection =
+        linkCatchList.stream()
+            .filter(item -> linkThrowList.contains(item))
+            .collect(Collectors.toList());
+
+    // check if intermediate throw and catch link event definitions are appears in pairs.
+    if (!(linkCatchEvents.size() == 0 && linkThrowEvents.size() == 0)) {
```

Comment:
Hi @skayliu. 
❌ As we can see the `Link Intermediate Event` defined in the bpmn specification, If there is no source link, the target link should be allowed to exist, we need to improve a bit here. 

(BPMN 2.0, chapter 10.4.4, page 253)
<img width="776" alt="image" src="https://user-images.githubusercontent.com/24605947/194682989-f200daf1-52d0-4023-9ae8-be7546b4909f.png">
--- 
Example:
<img width="554" alt="image" src="https://user-images.githubusercontent.com/24605947/194683718-2b703de8-801f-432b-9dbd-744f243f1006.png">
---
cc @saig0

Reply:
@lzgabel, @saig0, I've modified the link event validator and add more unit tests. It's now according the source link event to check if there is a matching target link event.

Reply:
@lzgabel, @saig0, I've modified the link event validator and add more unit tests. It's now according the source link event to check if there is a matching target link event.